drivers/syslog/syslog_device.c

```diff
@@ -453,7 +453,7 @@ static ssize_t syslog_dev_write(FAR struct syslog_channel_s *channel,
         {
           /* Check for pre-formatted CR-LF sequence */
 
-          if (remaining > 1 &&
+          if (remaining > 2 &&
```

Comment:
Did you test without this modification? I think it basically means "if remaining equal 2 or higher", with this modification now it is "if remaing equal 3 or higher". So, imagine the case where remaining is equal 2 and it is exactly (rare but can happen) : "\r\n", this test will fail.

Reply:
Ok. Let me check the case for "\r\n" separately

Reply:
I reviewed the logic and made a better solution